homeassistant/components/epion/sensor.py

```diff
@@ -0,0 +1,127 @@
+"""Support for Epion API."""
+from __future__ import annotations
+
+from homeassistant.components.sensor import (
+    SensorDeviceClass,
+    SensorEntity,
+    SensorStateClass,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import (
+    CONCENTRATION_PARTS_PER_MILLION,
+    PERCENTAGE,
+    UnitOfPressure,
+    UnitOfTemperature,
+)
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.device_registry import DeviceInfo
+from homeassistant.helpers.entity import generate_entity_id
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import DATA_API_CLIENT, DOMAIN
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Add an Epion entry."""
+    epion_base = hass.data[DOMAIN][entry.entry_id][DATA_API_CLIENT]
+
+    entities = []
+    current_data = epion_base.last_response
+    for epion_device in current_data["devices"]:
+        # Relevant keys are: deviceId, deviceName, locationId, lastMeasurement, co2, temperature, humidity, pressure
+        entities.append(EpionSensor(epion_base, epion_device, "co2", hass))
+        entities.append(EpionSensor(epion_base, epion_device, "temperature", hass))
+        entities.append(EpionSensor(epion_base, epion_device, "humidity", hass))
+        entities.append(EpionSensor(epion_base, epion_device, "pressure", hass))
+
+    async_add_entities(entities)
+
+
+class EpionSensor(SensorEntity):
+    """Representation of an Epion Air sensor."""
+
```

Comment:
New integration should use `has_entity_name = True`. Read more on https://developers.home-assistant.io/docs/core/entity#has_entity_name-true-mandatory-for-new-integrations

Reply:
Updated to use has_entity_name, I didn't know this was a new feature - definitely very useful!